tests/test_stl_binders.cpp

```diff
@@ -149,4 +188,9 @@ TEST_SUBMODULE(stl_binders, m) {
     m.def("get_vectorstruct", [] {
         return std::vector<VStruct>{{false, 5, 3.0, true}, {true, 30, -1e4, false}};
     });
+
+    py::bind_vector<RecursiveVector>(m, "RecursiveVector");
+    py::bind_map<RecursiveMap>(m, "RecursiveMap");
+    py::bind_map<MutuallyRecursiveMap>(m, "MutuallyRecursiveMap");
+    py::bind_vector<MutuallyRecursiveVector>(m, "MutuallyRecursiveVector");
```

Comment:
It seems like these need to go before this block:
```cpp
    // The rest depends on numpy:
    try {
        py::module_::import("numpy");
    } catch (...) {
        return;
    }
```
… to avoid the CI failures:
```

    def test_recursive_containers():
>       recursive_vector = m.RecursiveVector()
E       AttributeError: module 'pybind11_tests.stl_binders' has no attribute 'RecursiveVector'
```

Reply:
> It seems like these need to go before this block:

At first glance: that doesn't sound good. "Something else is probably wrong, somewhere."

Reply:
I don't think that there is a large problem. I just previously put the bindings in a place that was only active if numpy is available, so any test without numpy failed as a consequence. Should be fixed now.

Reply:
Yes, no problem at all: looking now at your latest commit, I misunderstood the before/after in your previous comment.